### PR TITLE
Doc2Vec update on negative sampling

### PR DESCRIPTION
**Pull request by nagireddyakshay**
- Updates -
. Fixed the test cases where negative sampling was set to 0 for Doc2Vec. Updated the value with 5(default value)

- Reason for the error -
We must set either 'hs' (hierarchical softmax) or 'negative' to be positive for proper training. When both 'hs=0' and 'negative=0', there will be no training. 

The cause for the above error being Doc2Vec do not update word embeddings if negative keyword is set to 0. So, in order to mitigate this, the contributors added a sanity check to the hs and negative arguments which checks if both hs and negative are set to 0 and throws the above error.

Here is the approved PR in the Gensim Library  for the above check- https://github.com/RaRe-Technologies/gensim/pull/3443

### CODE REVIEW

File: tests/test_doc2vec.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def det_hash(x):
 class Doc2VecTest(BaseTest):
 
     def test_fit_transform(self):
-        tw = TextWiser(Embedding.Doc2Vec(seed=1234, vector_size=2, min_count=1, workers=1, sample=0, negative=0,
+        tw = TextWiser(Embedding.Doc2Vec(seed=1234, vector_size=2, min_count=1, workers=1, sample=0, negative=5,
```
Comment (skadio) [inline tests/test_doc2vec.py:27]: I see that the test will pass, but having a zero here will still break TextWiser? 

Isn't there a validate_args() like function once these parameters are passed in and can be catch the 0 condition there to gracefully exit with an error message to say this is not possible. Then, you can still have a negative tests where you confirm/assert that we catch the invalid case.

Comment (nagireddyakshay) [reply tests/test_doc2vec.py:27]: Yes, we can do that. I believe we already have a validate function for the Doc2Vec class and we can extend its functionality to check for  unsupported keyword argument values as well!

Comment (bkleyn) [reply tests/test_doc2vec.py:27]: We actually don't do any validation of the arguments passed to Doc2Vec itself. We can of course add this one, but to validate all the arguments for Doc2Vec (and similarly other embedding/transformation methods) would be substantial effort.

Comment (skadio) [reply tests/test_doc2vec.py:27]: @bkleyn  you mean that this validation here https://github.com/fidelity/textwiser/blob/20c411596b7ae897a3bca83045221b4939ea3dbe/textwiser/options.py#L252 

is NOT validating the kwargs that go deep into the model, yes? And we stay away from "validating" arguments for downstream libraries and only validate our own arguments. 

This makes sense, as we possibly cannot check for everyhing. 

THEN, fixing the tests as above sounds like an easy solve. 

@nagireddyakshay  can you keep a simple test with that zero argument, which fails, add a decorator to skip that test so it does not break any pipeline, and inside the test, add your comments from this PR (the links you share to doc2vec etc..) Basically just narrate what we discussed here and what's happening here in that test for reference in future and explain our design decisions that, we know this fails but not asserting on purpose because it's in the downstream library etc.

Comment (nagireddyakshay) [reply tests/test_doc2vec.py:27]: Done. will add a test and update the PR. Thank you @skadio.
